app/components/offering-editor.js

```diff
@@ -0,0 +1,190 @@
+  room: null,
+
+  instructors: null,
+  instructorGroups: null,
+  learnerGroups: null,
+
+  filteredCohorts: computed('cohorts.[]', 'learnerGroups.[]', 'filter', function(){
+    let cohortProxy = ObjectProxy.extend({
+      selectedLearnerGroups: [],
+
+      hasAvailableLearnerGroups: notEmpty('filteredAvailableLearnerGroups'),
+
+      filter: '',
+
+      filteredAvailableLearnerGroups: computed('content.learnerGroups.[]', 'content.learnerGroups.@each.allDescendants.[]', 'selectedLearnerGroups.[]', 'filter', function(){
+        let defer = RSVP.defer();
+        let proxy = this;
+        let filter = proxy.get('filter');
+        let exp = new RegExp(filter, 'gi');
+
+        let activeGroupFilter = function(learnerGroup) {
+          let searchTerm = `${learnerGroup.get('title')}${learnerGroup.get('allParentsTitle')}`;
+
+          return (
+            learnerGroup.get('title') !== undefined &&
+            proxy.get('selectedLearnerGroups') &&
+            exp.test(searchTerm) &&
+            !proxy.get('selectedLearnerGroups').contains(learnerGroup)
+          );
+        };
+
+        this.get('content.topLevelLearnerGroups').then((cohortGroups) => {
+          let learnerGroups = [];
+          let promises = [];
+
+          cohortGroups.forEach((learnerGroup) => {
+            learnerGroups.pushObject(learnerGroup);
+
+            let promise = new Promise((resolve) => {
+              learnerGroup.get('allDescendants').then((descendants) => {
+                learnerGroups.pushObjects(descendants);
+                resolve();
+              });
+            });
+
+            promises.pushObject(promise);
+          });
+
+          all(promises).then(() => {
+            defer.resolve(learnerGroups.filter(activeGroupFilter).sortBy('sortTitle'));
+          });
+        });
+
+        return PromiseArray.create({
+          promise: defer.promise
+        });
+      }),
+    });
+
+    let cohorts = this.get('cohorts') ? this.get('cohorts') : [];
+
+    return cohorts.map((cohort) => {
+      let proxy = cohortProxy.create({
+        content: cohort,
+        selectedLearnerGroups: this.get('learnerGroups')
+      });
+
+      return proxy;
+    }).sortBy('title');
+  }),
+
+  actions: {
+    setOfferingType(value) {
+      this.set('singleOffering', value);
+    },
+
+    toggleMultiDay() {
+      this.set('isMultiDay', !this.get('isMultiDay'));
+    },
+
+    changeStartTime(date) {
+      let newStart = moment(date);
+      let startDate = moment(this.get('startDate'));
+
+      startDate.hour(newStart.format('HH'));
+      startDate.minute(newStart.format('mm'));
+      this.set('startDate', startDate.toDate());
+    },
+
+    changeEndTime(date) {
+      let newEnd = moment(date);
+      let endDate = moment(this.get('endDate'));
+
+      endDate.hour(newEnd.format('HH'));
+      endDate.minute(newEnd.format('mm'));
+      this.set('endDate', endDate.toDate());
+    },
+
+    addInstructorToBuffer(instructor){
```

Comment:
These method names *ToBuffer reflect the old pattern of creating a buffer object to send values to.  Seeing as how this component stores these values internally I think they can just be called addInstructor/addInstructorGroup for clarity.